config/v1/types.go

```diff
@@ -310,3 +310,90 @@ type DelegatedAuthorization struct {
+	DomainPatterns []string `json:"domainPatterns,omitempty"`
+
+	// maxAge is the delta time range in seconds during which hosts are regarded as HSTS hosts.
+	// If set to 0, it negates the effect, and hosts are removed as HSTS hosts.
+	// If set to 0 and includeSubdomains is specified, all subdomains of the host are also removed as HSTS hosts.
+	// maxAge is a time-to-live value, and if this policy is not refreshed on a client, the HSTS
+	// policy will eventually expire on that client.
+	// +required
+	MaxAge MaxAgePolicy `json:"maxAge"`
+
+	// preloadPolicy directs the client to include hosts in its host preload list so that
+	// it never needs to do an initial load to get the HSTS header (note that this is not defined
+	// in RFC 6797 and is therefore client implementation-dependent).
+	// +optional
+	PreloadPolicy PreloadPolicy `json:"preloadPolicy,omitempty"`
+
+	// includeSubDomainsPolicy means the HSTS Policy should apply to any subdomains of the host's
+	// domain name.  Thus, for the host bar.foo.com, if includeSubDomainsPolicy was set to RequireIncludeSubDomains:
+	// - the host app.bar.foo.com would inherit the HSTS Policy of bar.foo.com
+	// - the host bar.foo.com would inherit the HSTS Policy of bar.foo.com
+	// - the host foo.com would NOT inherit the HSTS Policy of bar.foo.com
+	// - the host def.foo.com would NOT inherit the HSTS Policy of bar.foo.com
+	// +optional
+	IncludeSubDomainsPolicy IncludeSubDomainsPolicy `json:"includeSubDomainsPolicy,omitempty"`
+}
+
+// MaxAgePolicy contains a numeric range for specifying a compliant HSTS max-age for the enclosing RequiredHSTSPolicy
+type MaxAgePolicy struct {
+	// The largest allowed value (in seconds) of the RequiredHSTSPolicy max-age
+	// default is one year (31536000 seconds)
+	// kubebuilder:validation:minimum=0:maximum=2147483647:default=31536000
+	LargestMaxAge *int32 `json:"largestMaxAge,omitempty"`
+
+	// The smallest allowed value (in seconds) of the RequiredHSTSPolicy max-age
+	// To set max-age=0 allows the deletion of an existing HSTS header from a host.  This is a necessary
+	// tool for administrators to quickly correct mistakes.
+	// default is one day (86400 seconds)
+	// kubebuilder:validation:minimum=0:maximum=2147483647:default=86400
+	SmallestMaxAge *int32 `json:"smallestMaxAge,omitempty"`
+}
+
+// PreloadPolicy contains a value for specifying a compliant HSTS preload policy for the enclosing RequiredHSTSPolicy
+// +kubebuilder:validation:Enum=RequirePreload;RequireNotPreload;NoOpinion
+type PreloadPolicy string
+
+const (
+	// requirePreloadPolicy means HSTS "preload" is required by the RequiredHSTSPolicy
+	RequirePreloadPolicy PreloadPolicy = "RequirePreload"
+
+	// requireNoPreloadPolicy means HSTS "preload" is forbidden by the RequiredHSTSPolicy
+	RequireNoPreloadPolicy PreloadPolicy = "RequireNotPreload"
+
+	// noOpinionPreloadPolicy means HSTS "preload" doesn't matter to the RequiredHSTSPolicy
+	NoOpinionPreloadPolicy PreloadPolicy = "NoOpinion"
```

Comment:
There are no JSON tags here, so might as well use the Go type names:
```suggestion
	// RequirePreloadPolicy means HSTS "preload" is required by the RequiredHSTSPolicy
	RequirePreloadPolicy PreloadPolicy = "RequirePreload"

	// RequireNoPreloadPolicy means HSTS "preload" is forbidden by the RequiredHSTSPolicy
	RequireNoPreloadPolicy PreloadPolicy = "RequireNotPreload"

	// NoOpinionPreloadPolicy means HSTS "preload" doesn't matter to the RequiredHSTSPolicy
	NoOpinionPreloadPolicy PreloadPolicy = "NoOpinion"
```